### PR TITLE
[chore] move solace receiver to generated lifecycle tests

### PR DESCRIPTION
**Pull request by atoulme**
Relates to #27849

### CODE REVIEW

**Comment by fatsheep9146** [comment]
I think failed checks are related to this change.

```
--- FAIL: Test_ComponentLifecycle (0.00s)
    --- FAIL: Test_ComponentLifecycle/traces-shutdown (0.00s)
        generated_component_test.go:67: 
            	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/solacereceiver/generated_component_test.go:67
            	Error:      	Received unexpected error:
            	            	receiver/solace/solacereceiver/receiver_status: cannot register view "receiver/solace/solacereceiver/receiver_status"; a different view with the same name is already registered
            	            	receiver/solace/solacereceiver/need_upgrade: cannot register view "receiver/solace/solacereceiver/need_upgrade"; a different view with the same name is already registered
            	            	receiver/solace/solacereceiver/receiver_flow_control_status: cannot register view "receiver/solace/solacereceiver/receiver_flow_control_status"; a different view with the same name is already registered
            	            	receiver/solace/solacereceiver/receiver_flow_control_recent_retries: cannot register view "receiver/solace/solacereceiver/receiver_flow_control_recent_retries"; a different view with the same name is already registered
            	Test:       	Test_ComponentLifecycle/traces-shutdown

```

File: receiver/solacereceiver/factory_test.go [hunk 1]
```diff
@@ -30,9 +30,11 @@ func TestCreateTracesReceiver(t *testing.T) {
 	require.NoError(t, err)
 	require.NoError(t, component.UnmarshalConfig(sub, cfg))
 
+	set := receivertest.NewNopCreateSettings()
+	set.ID = component.NewIDWithName("solace", "factory")
 	receiver, err := factory.CreateTracesReceiver(
 		context.Background(),
-		receivertest.NewNopCreateSettings(),
+		set,
```
Comment (dmitryax) [inline receiver/solacereceiver/factory_test.go:37]: Why is this needed?

Comment (atoulme) [reply receiver/solacereceiver/factory_test.go:37]: solace receiver registers a view in opencensus when it starts, based off the name of the receiver. The test generated uses the default name "", and that coincides with existing tests. Unfortunately, opencensus will return an error if you try to register the same metrics twice, so the workaround I found was to change the existing test to use a name so it would not clash with the generated test.

**Comment by atoulme** [comment]
I have resolved all comments, anything else pending to merge?

**Review by fatsheep9146 — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]
